csharp/src/Apache.Arrow/Arrays/BooleanArray.cs

```diff
@@ -34,11 +147,8 @@ public BooleanArray(ArrayData data)
 
         public override void Accept(IArrowArrayVisitor visitor) => Accept(this, visitor);
 
-        public bool? GetBoolean(int index)
+        public bool GetBoolean(int index)
         {
-            if (IsNull(index))
```

Comment:
Do we no longer support nulls in BooleanArrays?

Reply:
We kind of don't support nulls in any capacity at the moment. There's some infrastructure there, but most of it is missing for various array types.